### ISSUE TITLE
vscode does not show type hierarchy on maven multi module

### ISSUE THREAD

**Issue by clembo590** [redhat-developer/vscode-java#2871]
##### Environment
- Operating System: macos 13.0.1
- JDK version: 17
- Visual Studio Code version: Version: 1.74.2
- Java extension version:  java extension pack  v0.25.7

##### Steps To Reproduce
Steps To Reproduce
I have created a public repo so that the issue is easily reproducible:

step1: clone this repo:

```
git clone https://github.com/clembo590/issues.git --branch vscode-class-hierarchy-issue
```
step2: run mvn clean install
-->no issue

step 3: open the project in vscode

step4: open class called `First` and perform `show type hierarchy`

--> The error is that vscode does not show `Second` as a subclass of `First` AND IT SHOULD!

Step 5: move `Second.java` to the same folder than `First.java` and perform a `show type hierarchy` on `First.java`

--> This time the hierarchy is correct.

Type hierarchy works on a single maven module but not on multimodules.

Here is gif
![anim](https://user-images.githubusercontent.com/10899818/211784686-a23ee878-cf97-4bba-bc3c-2cdc07e9c32f.gif)

**Comment by CsCherrYY** [comment]
I can reproduce this issue in eclipse, looks like an upstream jdt issue.

![image](https://user-images.githubusercontent.com/45906942/211809971-9eb2a1ff-002f-4853-9795-94113fb379bf.png)

**Comment by rgrunber** [comment]
Seems similar to https://github.com/redhat-developer/vscode-java/issues/2219 . Maybe we can narrow it down more easily with this particular example ?

**Comment by snjeza** [comment]
I can't reproduce the issue in Eclipse JDT master
![jdt2871](https://user-images.githubusercontent.com/388609/212559132-82a5b2bb-ac23-4fb6-8a3c-efda6a95f4db.png)

It is reproducible in VS Code even with the jdt master..

**Comment by rgrunber** [comment]
It also works for me upstream. I tried with Eclipse 4.25 (2022-09) & 4.26 (2022-12) making sure to install the corresponding version of M2E support. I noticed 4.25 didn't recognize the project as Maven but did detect it on build (and added the correct decorations in the package explorer).

If it's reproducible in vscode-java every time, then let's start with that and see where things are going wrong.
